client/lib/gsuite/new-users.ts

```diff
@@ -160,30 +173,77 @@ const validateNewUserMailboxIsUnique = (
 /*
  * Adds a duplicate error to each mailBox with a duplicate mailbox
  */
-const validateNewUsersAreUnique = ( users: GSuiteNewUser[] ) => {
+const validateNewUsersAreUnique = ( users: GSuiteNewUser[] ): GSuiteNewUser[] => {
 	const mailboxesByCount: { [ mailbox: string ]: number } = countBy(
 		users.map( ( { mailBox: { value: mailBox } } ) => mailBox )
 	);
 
-	return users.map( ( { uuid, domain, mailBox, firstName, lastName } ) => ( {
+	return users.map( ( { uuid, domain, mailBox, firstName, lastName, password } ) => ( {
 		uuid,
 		firstName,
 		lastName,
 		domain,
 		mailBox: validateNewUserMailboxIsUnique( mailBox, mailboxesByCount ),
+		password,
 	} ) );
 };
 
 /*
- * Run all validations on a user:
- * domain - required
- * mailBox - required, validEmailCharacters, valid overall email
- * firstName - required, less than sixty characters
- * lastName - required, less than sixty characters
+ * Validates the specified password field.
+ *
+ * @see https://support.google.com/accounts/answer/32040 for requirements
+ */
+const validatePasswordField = ( { value, error }: GSuiteNewUserField ): GSuiteNewUserField => {
+	if ( ! error && 8 > value.length ) {
+		return {
+			value,
+			error: translate( "This field can't be shorter than %s characters.", { args: '8' } ),
+		};
+	}
+
+	if ( ! error && 100 < value.length ) {
+		return {
+			value,
+			error: translate( "This field can't be longer than %s characters.", { args: '100' } ),
+		};
+	}
+
+	if ( ! error && startsWith( value, ' ' ) ) {
+		return {
+			value,
+			error: translate( "This field can't start with a white space." ),
+		};
+	}
+
+	// Checks that passwords only have ASCII characters (see https://en.wikipedia.org/wiki/ASCII#Character_set)
+	const regexp = /[^\x20-\x7E]/;
+
+	if ( ! error && regexp.test( value ) ) {
+		const firstForbiddenCharacter = [ ...value ].find( ( character ) => regexp.test( character ) );
+
+		return {
+			value,
+			error: translate( "This field can't accept '%s' as character.", {
+				args: firstForbiddenCharacter,
+			} ),
+		};
+	}
+
+	if ( ! error && endsWith( value, ' ' ) ) {
+		return {
+			value,
+			error: translate( "This field can't end with a white space." ),
+		};
+	}
+
+	return { value, error };
```

Comment:
This only scratches the surface in terms of ensuring secure passwords. We can possibly handle that in further iterations, but this validation lets one get away with very weak combinations.

Reply:
Well the goal here was really to implement a validation mechanism in order to make sure that passwords match the rules dictated by the Google API. We can certainly look at providing users with a way to generate strong passwords in the future though.